gtsam/geometry/Cal3Bundler.cpp

```diff
@@ -99,18 +99,19 @@ Point2 Cal3Bundler::calibrate(const Point2& pi, OptionalJacobian<2, 3> Dcal,
   double x = (pi.x() - u0_) / fx_, y = (pi.y() - v0_) / fx_;
   const Point2 invKPi(x, y);
 
-  // initialize by ignoring the distortion at all, might be problematic for
-  // pixels around boundary
-  Point2 pn(x, y);
-
+  // initialize pn with distortion included
+  double px = pi.x(), py = pi.y(), rr = px * px + py * py;
+  double g = (1 + k1_ * rr + k2_ * rr * rr);
+  Point2 pn = invKPi / g;
```

Comment:
invKPi is in normalized coordinates but g is in pixel (unnormalized image) coordinates here. Shouldnt px and py be initialized to x and y respectively?

Reply:
If you change that, it becomes as though writing the first iteration of the loop outside the loop. 
So can the loop can be changed to a do-while? (I am not a fan of do-while or anything, but maybe this is a valid use case? )

Reply:
I did think about using a do-while yesterday. Let me update.

Reply:
I believe `g` should be initialized using the pixel coordinates since we capture the radial distortions that way. If `g` was initialized to `x` and `y` (aka the intrinsic coordinates), we wouldn't see any effect of the radial distortion in the inverse process.

Reply:
Great references. I agree that g should be defined using the intrinsic coordinates and not the image coordinates. I like your suggestion to use invKPi since that makes sense to me as per your references. I'll update the PR later tonight.

Reply:
@akshay-krishnan could you explain this point a bit more? I didn't quite follow

Reply:
John, could you check the references? They explain the pinhole camera model with distortion parameters. 
In this model, 

Pi = K * Pn_distorted where Pn_distorted = g * Pn_undistorted.
And g is a nonlinear function of Pn_undistorted. 

In calibrate() we are trying to find Pn_undistorted by optimization. The initial guess for Pn_undistorted = (invK * Pi) / g (using the above). But since g is a function of Pn_undistorted itself, we need an initial guess for g as well. The best we can do is initialize g using Pn_distorted ( = invK * Pi). 
I hope that makes sense.